schema.proto

```diff
@@ -0,0 +1,92 @@
+syntax = "proto3";
+
+enum WidgetType {
+  ais.custom = 0;
```

Comment:
I'm not sure I fully understand why the schema needs to be shipped to the client, could you explain that?

Reply:
This file (`schema.proto`) is only needed to generate the JavaScript counterpart (`schema.js`), I put it there for simplicity but it should live it's own project imho (since it is meant to be shared between the client and the server).

For `schema.js`, it is needed in the client so we can use it to generate the message and map all `$$type` and `$$params` to a number to keep the message as light as possible.